File: Code/Arrays_Bit_Manipulation/Container_With_Most_Water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0;
        int right = height.size() - 1;
        int maxWater = 0;

        while (left < right) {
            int heightMin = min(height[left], height[right]);
            int width = right - left;
            int currentArea = heightMin * width;
            maxWater = max(maxWater, currentArea);

            if (height[left] < height[right]) {
                ++left;
            } else {
                --right;
            }
        }

        return maxWater;
    }
};

File: Code/Arrays_Bit_Manipulation/Container_With_Most_Water.py
class Solution:
    def maxArea(self, height):
        left, right = 0, len(height) - 1
        max_water = 0
        while left < right:
            h = min(height[left], height[right])
            w = right - left
            max_water = max(max_water, h * w)
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
        return max_water

# Test case
if __name__ == "__main__":
    sol = Solution()
    height = [1, 8, 6, 2, 5, 4, 8, 3, 7]
    result = sol.maxArea(height)
    print("Maximum water that can be contained:", result)
